src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery.Test/ScenarioTests/AsrTestsBase.cs

```diff
@@ -31,33 +31,57 @@
 using Microsoft.WindowsAzure.Commands.Test.Utilities.Common;
 using RestTestFramework = Microsoft.Rest.ClientRuntime.Azure.TestFramework;
 
-namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.Test.ScenarioTests
+namespace RecoveryServices.SiteRecovery.Test
 {
     public abstract class AsrTestsBase : RMTestBase
     {
         protected string vaultSettingsFilePath;
-        private readonly ASRVaultCreds asrVaultCreds;
+        protected string powershellFile;
+        private ASRVaultCreds asrVaultCreds;
         private CSMTestEnvironmentFactory csmTestFactory;
-        private readonly EnvironmentSetupHelper helper;
+        private EnvironmentSetupHelper helper;
 
         protected AsrTestsBase()
         {
-            this.vaultSettingsFilePath = Path.Combine(
-                AppDomain.CurrentDomain.BaseDirectory,
-                "ScenarioTests\\vaultSettings.VaultCredentials");
+        }
 
+        protected void initialize()
+        {
             if (File.Exists(this.vaultSettingsFilePath))
             {
                 try
-                {
-                    var serializer1 = new DataContractSerializer(typeof(ASRVaultCreds));
-                    using (var s = new FileStream(
-                        this.vaultSettingsFilePath,
-                        FileMode.Open,
-                        FileAccess.Read,
-                        FileShare.Read))
+                {  
+                    if (File.ReadAllText(this.vaultSettingsFilePath).ToLower().Contains("<asrvaultcreds"))
```

Comment:
@viverm to read from/write to files, use `DataStore.ReadFromFileStream()` from in the `FileUtilities` class in `Commands.Common`

https://github.com/Azure/azure-powershell/blob/preview/src/Common/Commands.Common/Utilities/FileUtilities.cs#L41

Reply:
Used FileUtilities.DataStore.ReadFileAsText method